cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -111,4 +137,112 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
       responder.sendString(HttpResponseStatus.OK, prevInProgressUUID);
     }
   }
+
+  /**
+   * Get the list of bundle status from status.json.
+   *
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles")
+  public void listSupportBundles(HttpRequest request, HttpResponder responder) throws Exception {
+    // ensure the user has authentication to create supportBundle
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.LIST);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    if (!baseDirectory.exists()) {
+      LOG.debug("No content in Support Bundle.");
+      return;
+    }
+    List<SupportBundleOperationStatus> bundleOperationStatusList = new ArrayList<>();
+    DirUtils.listFiles(baseDirectory)
+      .stream()
+      .filter(file -> !file.isHidden() && file.isDirectory())
+      .forEach(uuidFile -> {
+        SupportBundleOperationStatus bundleOperationStatus = null;
+        try {
+          bundleOperationStatus = bundleGenerator.getBundle(uuidFile.getName());
+        } catch (IOException e) {
+          LOG.debug("Can not find status json: {}", uuidFile.getName());
+        }
+        bundleOperationStatusList.add(bundleOperationStatus);
+      });
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatusList));
+  }
+
+  /**
+   * Get the specific bundle status from status.json by using uuid.
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles/{uuid}/status")
+  public void getSupportBundleStatus(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    SupportBundleOperationStatus bundleOperationStatus = bundleGenerator.getBundle(uuid);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatus));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundles/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.DELETE);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    File uuidFile = new File(baseDirectory, uuid);
```

Comment:
nit: Please move the uuidFile implementation to a helper method (i.e. something like):
```
private File getUUIDFile(String uuid) {
  File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
  return new File(baseDirectory, uuid);
}
```